writer_test.go

```diff
@@ -189,3 +233,101 @@ func readPartition(topic string, partition int, offset int64) (msgs []Message, e
 		msgs = append(msgs, msg)
 	}
 }
+
+func testWrtierBatchBytes(t *testing.T) {
```

Comment:
I feel like we're missing a condition here.  We're ensuring that the two messages are sent in a batch, but I'd also like to see the case where they're written in two separate requests (e.g. if `MaxMessageBytes` were set to `47` so that we cover the `lastMsg != nil` condition).

Reply:
Case added!